### PR TITLE
Command to toggle issuing AoC completionist

### PR DESCRIPTION
**Pull request by ChrisLovering**
Labeled as **DO NOT MERGE** as I will need to migrate redis caches to the new namespaces before merging this.

This adds a new command, and redis cache, to control whether the AoC completionist role is issued.

This also refactors the cog a little it while I was there. As usual, this PR is best reviewed commit-by-commit.

### CODE REVIEW

**Review by janine9vn — CHANGES_REQUESTED** [review]
If the role from the loaded in config can't be found, then the `&aoc ct` to do the initial toggle-on fails. I think it would be better to output the error somewhere visible for admins/the Event Runner to see if this occurs. It does get logged, but it wouldn't be obvious for a few minutes that something was wrong and to check the logs. I think outputting to devlog like we do for leaderboard errors makes sense.

Otherwise everything works well and as expected. I wasn't able to test the scheduler changes, but I can also worry about it in 11 months.

**Comment by ChrisLovering** [comment]
> If the role from the loaded in config can't be found, then the `&aoc ct` to do the initial toggle-on fails. I think it would be better to output the error somewhere visible for admins/the Event Runner to see if this occurs. It does get logged, but it wouldn't be obvious for a few minutes that something was wrong and to check the logs. I think outputting to devlog like we do for leaderboard errors makes sense.
> 
> Otherwise everything works well and as expected. I wasn't able to test the scheduler changes, but I can also worry about it in 11 months.

It's a warning log that will be pushed to sentry, and output to #dev-alerts, like most other problems like this in our other bots

**Review by janine9vn — APPROVED** [review]

**Review by shtlrs — APPROVED** [review]

File: bot/exts/advent_of_code/_cog.py [hunk 1]
```diff
@@ -137,6 +138,26 @@ async def adventofcode_group(self, ctx: commands.Context) -> None:
         if not ctx.invoked_subcommand:
             await self.bot.invoke_help_command(ctx)
 
+    @with_role(Roles.admins, fail_silently=True)
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/exts/advent_of_code/_cog.py:141]: Yea toggling twice was my thought, since it's rarely going to happen anyway.

**Comment by ChrisLovering** [comment]
Just realised I never gave event leads access to run this command, added a small commit to fix this.

**Pull request merged**